guide.js

```diff
@@ -363,47 +351,53 @@ function updateIframe(visible){
+		let sidebar = document.getElementById("sidebar");
+		let divider = document.getElementById("dragMe");
+		let mainPanel = document.getElementsByClassName("mainPanel")[0]; //TODO: homogenize the id for this div on each webpage. //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
+		if(sidebar != null && divider != null && mainPanel != null){
 			sidebar.style.display = "none";
+			divider.style.display = "none";
+			mainPanel.style.width = "100%"; //Want this to be 100% width since draggable will set it assuming there is a sidePanel even if there isn't.
 		}
-
-		if(settings.iframeCheck == "window"){
-			var links = document.getElementsByTagName("A");
+		//update links to redirect to desired place based on settings.
+		if(settings.iframeCheck == "window"){ //If the user wants links to redirect to a second window.
+			var links = document.getElementsByClassName("mainPanel")[0].getElementsByTagName("A"); //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
 			for(var lnk of links){
-				if(lnk.target == "_blank"){
-					lnk.target = "myframe";
+				if(lnk.target == "_blank" || lnk.target == "myframe"){
+					lnk.target = "externalSecondWindow";
 				}
 			}
 		}
-		else{
-			//reset links to open in new tab, otherwise it looks like they're doing nothing.
-			var links = document.getElementsByTagName("A");
+		else{//iframeCheck setting is "off" and the user doesn't specfically want it to open in a second window, so we do new tab.
+			var links = document.getElementsByClassName("mainPanel")[0].getElementsByTagName("A"); //DOES THIS GRAB OTHER GUIDE FRAMES FROM WITHIN THE IFRAME? LOWER SCOPE TO A GUIDE ID?
 			for(var lnk of links){
-				if(lnk.target == "myframe"){
+				if(lnk.target == "myframe" || lnk.target == "externalSecondWindow"){
 					lnk.target = "_blank";
 				}
 			}
 		}
 		__displayingIframe = false;
 	}
+	__linkFormatType = settings.iframeCheck;
 }
 
+//The root function of the iframe resizing settings, runs each time the window size changes.
 var windowResizeId = null;
 function onWindowResize(event){
 	//on window resize, we may want to hide sidebar.
-	//only resize after being still for 50ms
+	//only resize after being still for 10ms
 	if(windowResizeId != null){
 		clearTimeout(windowResizeId);
 	}
-	windowResizeId = setTimeout(checkIframeSize,50,event);
+	windowResizeId = setTimeout(checkIframeSize,10,event);
 }
 
+//Checks the window size and pass a true/false to the updateIframe function to indicate if the iframe should be turned on/off.
 function checkIframeSize(event){
 	windowResizeId = null;
 	if(settings?.iframeCheck == "on" || 
-	(settings?.iframeCheck == "auto" && window.innerWidth >= settings.iframeMinWidth)){
+	(settings?.iframeCheck == "auto" && !window.matchMedia("(max-width: " + settings.iframeMinWidth + "px)").matches)){ //if iframe setting is on or set to auto and window is larger than setting for min width.
```

Comment:
is there a reason for this change?

Reply:
I believe in my PR I said I changed it because "matchMedia is rad as hell" but other than that I really have no other justification for this change, it was just something I learned was possible and since it worked when I tested it, I just left it in lol. Does make the code less readable?

I tried changing it back to the other conditional and it is identical. I've decided for now to revert it for readability.

Reply:
Worth noting, zooming in/out changes the window.innerWidth value, but it appears to also change it with the window.matchMedia approach. Ideally I'll find a solution that can fix this small bug.

Reply:
This bug is extremely minor and people with super zoomed in views that really want to keep the sidepanel can just use the always on setting for now. Going to resolve this one.